chtransmissionformat/README.md

```diff
@@ -0,0 +1,267 @@
+
+- `CHMED` (Prefix): This is the fixed string `CHMED`.
+  It is always the same value and should be used to determine if the given input string
+  should be tried to parse as ChTransmissionFormat.
+  During parsing, it should be treated as case-insensitive (e.g. `ChMed` should also be accepted)
+  but during serialization all uppercase should be used.
+- `ReleaseYear`: two digits (e.g. `16`)
+- `SubVersion`: At least a single uppercase alphabetic character (e.g. `A`).
+  Depending on the `ReleaseYear`, more than one character can be allowed.
+  During parsing, it should be treated as case-insensitive (e.g. `a` should also be accepted)
+  but during serialization all uppercase should be used.
+- `AdditionalMetaData`: Based on the `ReleaseYear`, additional meta data may be required.
+- `Data`: The actual data (JSON) which (after applying transformations based on the `ReleaseYear`)
+  needs to conform with the corresponding specification defining the format of the JSON.
+
+Since part of the format is dependent on the `ReleaseYear`,
+each available `ReleaseYear` is covered independently.
+Only years listed here are considered valid for use with ChTransmissionFormat.
+All constraints listed above also apply to the specific parts defined for a `ReleaseYear`.
+
+Note that the JSON format itself does not contain a version marker
+as it would be duplicated data when transmitted using ChTransmissionFormat
+and one of the goals is to reduce the data size.
+Therefore, CHMED documents should always be transmitted using ChTransmissionFormat.
+
+### ReleaseYear 16
+
+```
+CHMED16{SubVersion}{Compression}{Data}
+```
+
+Representation as regex:
+
+```regex
+^CHMED16([A-Z])([0-1])(.*)$
+```
+
+Description and constraints:
+
+- `SubVersion`: A single uppercase alphabetic character.
+- `Compression`: This is the only part of `AdditionalMetaData`.
+  It is a single digit which either is `0` or `1`
+  and determines if compression is used for the `Data`.
+- `Data`:
+  - if `Compression = 0`: Plain JSON document.
+    Note that it is highly recommended to always compress the contents.
+    Later versions (e.g. `ReleaseYear` 23) do no longer support documents without compression.
+  - if `Compression = 1`: Base64 encoded and gzipped JSON document
+
+**Example**
+
+```
+CHMED16A1H4sIAAAAAAAEAMVU3W7TMBR+lcq3S4SPHdtx7raVAaKFqutAAnoREreJ2jpT4gKj6ptxx4txnCwVSKQSu0GVqvPX7+fIpwdyuXcFSYiSFCjloGKtNQnI2GGRUZAh1SGwBUASyYTqC8oSSnHgVe4HZM7z1UqF6WcqwkhxjCKdhcJkWc650GIlcXZq8sXDvSEJtHGZpTtjXUOSj4cORyuIpWpRu0EekFnVDYzxmwbtZ+l13dTV7g9t5Ij1eZVicfYWMRab9byxmF1urMH8zpbezO3iNTkGj4RRJIHH7Bwj/J2RhUAHGa+2e/fF1PnPH9bu7XqAnHEhZHzGrqeGJ5EXVVbk9T7bfHr20tTfBwRAxCQHen7f/y7guqi2pnGmLm1j7MbUA/Qx1VLz8/aftvvf7L9Pm2ZQAQgBTEX/QwL+dJa6Et8/SQ7kqr0z0DGEFPxbDsh16R48lqktZjdv8FYwnabfyl2KhRfG5giasIBMTj3nWp4JPrmEjJ93J+fxx5PGTY3X1tbSzuHpGDthCPUu3eKM4O2Kujbr29C3QcSd9ex0udibYyKUWvbLZX3Au54EdupFfSD6QD4CCBEIBcsjjpJZUVnv6yJCxlEsYMRk+59z62pj/MLuLBreoe2vZj2KsfOhvMcyp0yhAzLfbbzj4y8tzloh3gQAAA==
```

Comment:
Might be nice to see include an uncompressed example as well

Reply:
Would you recommend this despite this sentence (this is the updated version but existed in the first draft as well)?

> Since it is RECOMMENDED to compress the data, no examples are provided with `Compression = 0`.

Reply:
This is further down so I hadn't read it at this time. I'm not sure what I would do. Putting myself into the shoes of a dev that needs to implement a (feature complete) parser for this format I would appreciate an official example to test my implementation with.

Reply:
Good point, it would not make sense to suggest to not even parse that since it is part of the specification. I added an example and also extended the parsing example to include `Compression = 0`.